### PR TITLE
refactor: clean model tree item

### PR DESCRIPTION
**Pull request by liuxi150**
## Summary by CodeRabbit

- **New Features**
	- Introduced the `ModelTreeComponent` class for enhanced 3D scene tree functionality.

- **Bug Fixes**
	- Improved handling of transformation data and initialization processes in the `VFXItem` class.
	- Enhanced error handling with more descriptive messages for JSON and binary data loading failures.

- **Refactor**
	- Removed several outdated methods and classes related to tree structures and animations, streamlining the codebase.

- **Documentation**
	- Updated documentation for clarity on the `parent` property in the `VFXItem` class.

### CODE REVIEW

File: plugin-packages/model/src/gltf/json-converter.ts [hunk 1]
```diff
@@ -43,6 +43,11 @@ export class JSONConverter {
     const oldBinUrls = oldScene.bins ?? [];
     const binFiles: ArrayBuffer[] = [];
 
+    //@ts-expect-error
+    if (sceneJSON.version >= '3.0') {
```
Comment (yiiqii) [inline plugin-packages/model/src/gltf/json-converter.ts:47]: 版本号不能这么粗暴的判断，比如 "10.0">="3.0" 就是 false
```
const v = sceneJSON.version.split('.');
if((Number(v[0])===3 && Number(v[1])===0)||Number(v[0])>3)
```

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: plugin-packages/model/src/plugin/model-tree-component.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import type { Engine } from '@galacean/effects';
+import { Behaviour, effectsClass, spec } from '@galacean/effects';
+import type { ModelTreeContent } from '../index';
+
+/**
+ * 插件场景树组件类，实现 3D 场景树功能
+ *
+ * FIXME: 有些发布的新JSON包含TreeComponent，这里做兼容处理，否则会报错
+ * @since 2.0.0
```
Comment (yiiqii) [inline plugin-packages/model/src/plugin/model-tree-component.ts:9]: `@since 2.1.0`

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: plugin-packages/model/src/gltf/json-converter.ts (hunk 1 above)
Comment (liuxi150) [reply plugin-packages/model/src/gltf/json-converter.ts]: 已修改

File: plugin-packages/model/src/plugin/model-tree-component.ts (hunk 2 above)
Comment (liuxi150) [reply plugin-packages/model/src/plugin/model-tree-component.ts]: 已修改

**Review by wumaolinmaoan — APPROVED** [review]

**Review by yiiqii — APPROVED** [review]

**Pull request merged**